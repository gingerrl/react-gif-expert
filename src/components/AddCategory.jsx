import { useState } from "react";
import PropTypes from 'prop-types'
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState();
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setInputValue("");

    onNewCategory(inputValue.trim())
    // setCategories((categories) => [inputValue, ...categories]);
  };
  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue || ''}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};


AddCategory.propTypes={
  onNewCategory: PropTypes.func.isRequired
}