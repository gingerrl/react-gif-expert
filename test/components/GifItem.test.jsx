import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const title = "Saitema";
  const url = "https://one-punch/";

  test("debe hacer match con snapshot ", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el Ur; y el Alt indicado ", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole('img').src).toBe(url)
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("debe mostar el titulo en el componente ", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
