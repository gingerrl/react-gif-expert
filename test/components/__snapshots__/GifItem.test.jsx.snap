// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe hacer match con snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitema"
      src="https://one-punch/"
    />
    <p>
      Saitema
    </p>
  </div>
</div>
`;
